firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/TesterSignInClient.java

```diff
@@ -0,0 +1,185 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.appdistribution;
+
+import static com.google.firebase.appdistribution.FirebaseAppDistributionException.Status.AUTHENTICATION_CANCELED;
+import static com.google.firebase.appdistribution.FirebaseAppDistributionException.Status.AUTHENTICATION_FAILURE;
+
+import android.app.Activity;
+import android.app.AlertDialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.content.pm.ResolveInfo;
+import android.net.Uri;
+import android.util.Log;
+import androidx.annotation.NonNull;
+import androidx.browser.customtabs.CustomTabsIntent;
+import com.google.android.gms.tasks.CancellationTokenSource;
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.android.gms.tasks.Task;
+import com.google.android.gms.tasks.TaskCompletionSource;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.installations.FirebaseInstallationsApi;
+import java.util.List;
+
+public class TesterSignInClient {
+  private static final String TAG = "FADSignInTester";
+
+  private TaskCompletionSource<Void> signInTaskCompletionSource = null;
+  private CancellationTokenSource signInCancellationSource;
+  private final String SIGNIN_REDIRECT_URL =
+      "https://appdistribution.firebase.google.com/pub/testerapps/%s/installations/%s/buildalerts?appName=%s&packageName=%s";
+  private final FirebaseApp firebaseApp;
+  private final FirebaseInstallationsApi firebaseInstallationsApi;
+
+  TesterSignInClient(
+      @NonNull FirebaseApp firebaseApp,
+      @NonNull FirebaseInstallationsApi firebaseInstallationsApi) {
+    this.firebaseApp = firebaseApp;
+    this.firebaseInstallationsApi = firebaseInstallationsApi;
+  }
+
+  @NonNull
+  public Task<Void> signInTester(@NonNull Activity currentActivity) {
```

Comment:
this is a nit but since you have the current activity, you don't need the firebase app. you can get the application context using `activity.getApplicationContext()` and package name using `activity.getPackageName()`

Reply:
I think I need the FirebaseApp to read the gmpappid to construct the sign in url. But I could pass that in as a parameter instead